packages/web3-eth/src/types.ts

```diff
@@ -8,135 +8,89 @@ import {
 	HexString32Bytes,
 	HexString256Bytes,
 	HexStringBytes,
-	Uint,
 } from 'web3-utils';
 
-export enum ChainNames {
-	MAINNET = 'mainnet',
-	GOERLI = 'goerli',
-	KOVAN = 'kovan',
-	RINKEBY = 'rinkeby',
-	ROPSTEN = 'ropsten',
+export type chain = 'goerli' | 'kovan' | 'mainnet' | 'rinkeby' | 'ropsten' | 'sepolia';
+export type hardfork =
+	| 'arrowGlacier'
+	| 'berlin'
+	| 'byzantium'
+	| 'chainstart'
+	| 'constantinople'
+	| 'dao'
+	| 'homestead'
+	| 'istanbul'
+	| 'london'
+	| 'merge'
+	| 'muirGlacier'
+	| 'petersburg'
+	| 'shanghai'
+	| 'spuriousDragon'
+	| 'tangerineWhistle';
+
+export interface CustomChain<NumberType = Numbers> {
+	name?: string;
+	networkId: NumberType;
+	chainId: NumberType;
 }
 
-export enum HardForks {
-	CHAIN_START = 'chainstart',
-	HOMESTEAD = 'homestead',
-	DAO = 'dao',
-	TANGERINE_WHISTLE = 'tangerineWhistle',
-	SPURIOUS_DRAGON = 'spuriousDragon',
-	BYZANTIUM = 'byzantium',
-	CONSTANTINOPLE = 'constantinople',
-	PETERSBURG = 'petersburg',
-	ISTANBUL = 'istanbul',
-	BERLIN = 'berlin',
-	LONDON = 'london',
+export interface Common<NumberType = Numbers> {
+	customChain: CustomChain<NumberType>;
+	baseChain?: chain;
+	hardfork?: hardfork;
 }
 
-export interface Transaction {
+export interface Transaction<NumberType = Numbers> {
 	from?: Address;
 	to?: Address;
-	value?: Numbers;
-	gas?: Numbers;
-	gasPrice?: Numbers;
-	type?: Numbers;
-	maxFeePerGas?: Numbers;
-	maxPriorityFeePerGas?: Numbers;
+	value?: NumberType;
+	gas?: NumberType;
+	gasPrice?: NumberType;
+	type?: NumberType;
+	maxFeePerGas?: NumberType;
+	maxPriorityFeePerGas?: NumberType;
```

Comment:
I am not fan for this approach. This approach becomes very open cause problems while using types. e.g. a user can specify `Error` as `NumberType` that obviously can't be valid. 

Ideal solution would be that `Numbers` type should be self aware what types it can be represented and then you pass generic to it with that given constraints.

Reply:
I believe this solves the issue?

```typescript
export interface Transaction<NumberType extends Numbers = Numbers> { ... }
```